src/briefcase/integrations/download.py

```diff
@@ -0,0 +1,88 @@
+from cgi import parse_header
+from urllib.parse import urlparse
+
+import requests
+
+from briefcase.exceptions import (
+    BadNetworkResourceError,
+    MissingNetworkResourceError,
+    NetworkFailure,
+)
+
+
+class Download:
```

Comment:
I promoted `download_file` to a full fledged tool. While it may be a bit overkill in its current form, I think the consistency is better and this will make extending it easier. For instance, adding hash checking or including methods such a `zip` or `tarball` so unpacking could be done here as well (although, I acknowledge scope arguments against doing that within a "download" class).